File: angularfrontend/src/app/modules/product.ts
// import {Category} from './category';
// import {Promotion} from './promotion';
// import {MyBrand} from '../enums/myBrand';
// import {MyGeneration} from '../enums/myGeneration';

// export interface Product {
//   productId: number,
//   productName: string,
//   brand: MyBrand,
//   model: string,
//   price: number,
//   description: string,
//   image: Blob,
//   category: Category;
//   promotion: Promotion;
// }


import {Category} from './category';
import {Promotion} from './promotion';
import {MyBrand} from '../enums/myBrand';
import {MyGeneration} from '../enums/myGeneration';

export interface Product {
  productId: number,
  productName: string,
  brand: MyBrand,
  model: string,
  price: number,
  description: string,
  image: Blob,
  category: Category;
}

File: jwt-youtube-master/src/main/java/com/youtube/jwt/entity/Product.java
package com.youtube.jwt.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;

import com.youtube.jwt.enumType.MyBrand;

import java.util.Arrays;
import java.util.List;

@Data
@ToString
@Entity
public class Product {

    @Id
    @GeneratedValue
    private Long productId;

    private String productName;

    @Enumerated(EnumType.STRING)
    private MyBrand brand;

    private String model;

    private float price;

    private String description;

    @Column(name = "image", length = 1000000)
    private byte[] image;

    @ManyToOne(cascade = {CascadeType.DETACH , CascadeType.PERSIST
            , CascadeType.MERGE , CascadeType.REFRESH})
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToOne(cascade = {
            CascadeType.PERSIST , CascadeType.MERGE,
            CascadeType.DETACH , CascadeType.REFRESH
    })
    @JoinColumn(name = "promotion_id")
    private Promotion promotion;




}
